spec/models/employer_spec.rb

```diff
@@ -1,17 +1,27 @@
 require "rails_helper"
 
 describe Employer do
-  let(:name) { "My Company" }
-  let(:account_number) { 1234567 }
-  let(:employer) { FactoryBot.build_stubbed(:employer) }
-
   describe "initialisation" do
-    it "has an account number" do
-      expect(employer.account_number).to eq(account_number)
+    describe "valid case" do
+      it "has an account number" do
+        employer = create(:employer, account_number: 1234567)
+
+        expect(employer.account_number).to eq(1234567)
+      end
+
+      it "has an employer name" do
+        employer = create(:employer, name: "My Company")
+
+        expect(employer.name).to eq("My Company")
+      end
```

Comment:
Rather than testing that the attributes are assigned, I try and test the behavior.

For example, when the correct attributes are passed, it is valid:

```ruby
context "when all the correct attributes are assigned" do
  it "is valid" do
    employer = build_stubbed(:employer, account_number: 1234567, name: "Test company")
    
    expect(employer).to be_valid
  end
end
```